geniza/corpus/metadata_export.py

```diff
@@ -189,3 +189,85 @@ class PublicDocumentExporter(DocumentExporter):
 
     def get_queryset(self):
         return super().get_queryset().filter(status=Document.PUBLIC)
+
+
+class FragmentExporter(Exporter):
+    """
+    A subclass of :class:`geniza.common.metadata_export.Exporter` that
+    exports information relating to :class:`~geniza.corpus.models.Fragment`.
+    """
+
+    model = Fragment
+    csv_fields = [
+        "shelfmark",
+        "pgpids",
+        "old_shelfmarks",
+        "collection",
+        "library",
+        "library_abbrev",
+        "collection_name",
+        "collection_abbrev",
+        "url",
+        "iiif_url",
+        "is_multifragment",
+        "created",
+        "last_modified",
+    ]
+
+    def get_queryset(self):
+        """
+        Applies some prefetching to the base Exporter's get_queryset functionality.
+
+        :return: Custom-given query set or query set of all documents
+        :rtype: QuerySet
+        """
+        return (
+            super()
+            .get_queryset()
+            .select_related("collection")
+            .prefetch_related("documents")
+        )
+
+    def get_export_data_dict(self, fragment):
+        data = {
+            "shelfmark": fragment.shelfmark,
+            "pgpids": [doc.pk for doc in fragment.documents.all()],
+            "old_shelfmarks": fragment.old_shelfmarks,
+            "url": fragment.url,
+            "iiif_url": fragment.iiif_url,
+            "is_multifragment": fragment.is_multifragment,
+            "created": fragment.created,
+            "last_modified": fragment.last_modified,
+        }
+        # it's possible (although unlikely) for collection to be unset
+        if fragment.collection:
+            # NOTE: this results in a lot of redundant info;
+            # maybe collection shortname is enough for fragment csv?
+            data.update(
+                {
+                    "collection": fragment.collection,
+                    "library": fragment.collection.library,
+                    "library_abbrev": fragment.collection.lib_abbrev,
+                    "collection_name": fragment.collection.name,
+                    "collection_abbrev": fragment.collection.abbrev,
+                }
+            )
+        return data
```

Comment:
Hm. Will these fragment data be included in the metadata exports? At least any redundant values there will be guaranteed to match. I suppose the redundancy in this code snippet here is also guaranteed? Since it's just a matter of pulling additional attributes off the `fragment.collection` object.

Reply:
My plan was to include this alongside the documents export — there's overlapping information, but for documents that run across multiple fragments I think it may be easier to work with this vs. trying to split out multiple parallel fields in the document csv.

After seeing this, I'm not sure how valuable it is to have all the collection details on each row. I wonder if the short name is enough if there is a lookup somewhere that you can reference if you're not familiar with the collections — but also not sure I want to go to the effort of making a collections.csv to sit next to this as a reference.

Reply:
Yeah, agreed, not sure it's worth an extra csv file

Reply:
I'll ask if they want the extra collection fields when we have the team to acceptance testing